coursefunctions.h

```diff
@@ -429,13 +465,39 @@ void TestVertexShader(Buffer2D<PIXEL> & target)
          *********************************************/
         // Your rotation code that integrates with 'colorUniforms', used by 'myColorVertexShader' goes here
 
+        colorUniforms = Attributes();
+
+        // Clear the Attributes
+        colorUniforms.insertDbl(1); // Scale
+        colorUniforms.insertDbl(1);
+        colorUniforms.insertDbl(1);
+        colorUniforms.insertDbl(0); // Rotate
+        colorUniforms.insertDbl(0);
+        colorUniforms.insertDbl(30);
+        colorUniforms.insertDbl(0); // Translate
+        colorUniforms.insertDbl(0);
+        colorUniforms.insertDbl(0); 
+
         DrawPrimitive(TRIANGLE, target, colorTriangle, colorAttributes, &colorUniforms, &myColorFragShader, &myColorVertexShader);
 
         /*************************************************
          * SCALE-TRANSLATE-ROTATE in left-to-right order
          * the previous transformations concatenated.
          ************************************************/
-		// Your scale-translate-rotation code that integrates with 'colorUniforms', used by 'myColorVertexShader' goes here
+        // Your scale-translate-rotation code that integrates with 'colorUniforms', used by 'myColorVertexShader' goes here
+
+        // Clear the Attributes
+        colorUniforms = Attributes();
+
+        colorUniforms.insertDbl(.5); // Scale
```

Comment:
Looks like you got your operator overloaded and working, but may I suggest making a class that with functions that will make the transforming easier? A class with functions that you can call and pass in x y z values, that will set up the matrix in a way that is intuitive. Just a thought. Otherwise, good job!